# HG changeset patch
# User Shai Ayal <shaiay@users.sourceforge.net>
# Date 1271272534 -10800
#      Wed Apr 14 22:15:34 2010 +0300
# Node ID 6538ff56294975865bb7e285f9f294fa5aa73b4a
# Parent  4d1fc073fbb72ced78512a1012360eba5327ee73
document keypress and mouse callback properties

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,12 @@
+2010-04-14  Shai Ayal  <shaiay@users.sourceforge.net>
+
+	* NEWS: Update.
+
 2010-04-08  Marco Atzeri  <marco_atzeri@yahoo.it>
 
 	* Makefile.am (make-version-links, remove-version-links):
 	Use SCRIPTS_EXE_SUFFIX for scripts, not EXEEXT.
 
 2010-04-07  John W. Eaton  <jwe@octave.org>
 
 	* README.ftp, README.devel:
diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -1,11 +1,17 @@
 Summary of important user-visible changes for version 3.3:
 ---------------------------------------------------------
 
+ ** The fltk graphics backend now implements the following callback
+    properties: keypressfcn, keyreleasefcn, windowbuttondownfcn,
+    windowbuttonmotionfcn, windowbuttonupfcn, and currentpoint. These
+    enable keyboard and mouse interaction, an example of which can be
+    seen in the __fltk_ginput__.m script.
+
  ** BLAS and LAPACK libraries are now required to build Octave.  The
     subset of the reference BLAS and LAPACK libraries has been removed
     from the Octave sources.
 
  ** The `lookup' function was extended to be more useful for
     general-purpose binary searching.  Using this improvement, the
     ismember function was rewritten for significantly better
     performance.
diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,13 @@
+2010-04-14  Shai Ayal  <shaiay@users.sourceforge.net>
+
+	* interpreter/plot.txi: Document keypress and mouse callback
+	properties
+
 2010-04-07  John W. Eaton  <jwe@octave.org>
 
 	* interpreter/testfun.txi: Document runtests.
 
 2010-03-24  John W. Eaton  <jwe@octave.org>
 
 	* interpreter/bugs.txi: Update for bug tracker.
 
diff --git a/doc/interpreter/plot.txi b/doc/interpreter/plot.txi
--- a/doc/interpreter/plot.txi
+++ b/doc/interpreter/plot.txi
@@ -934,17 +934,17 @@ beingdeleted           hittest          
 busyaction             integerhandle          pointershapecdata      windowbuttonmotionfcn
 buttondownfcn          interruptible          pointershapehotspot    windowbuttonupfcn
 children               inverthardcopy         position               windowbuttonwheelfcn
 clipping               keypressfcn            renderer               windowstyle
 closerequestfcn        keyreleasefcn          renderermode           wvisual
 color                  menubar                resize                 wvisualmode
 colormap               mincolormap            resizefcn              xdisplay
 createfcn              name                   selected               xvisual
-current_point          nextplot               selectionhighlight     xvisualmode
+currentpoint          nextplot               selectionhighlight     xvisualmode
 currentaxes            numbertitle            selectiontype
 currentcharacter       paperorientation       tag
 @end verbatim
 @end group
 @end smallformat
 @c @end example
 @c @end flushleft
 @c @example
@@ -1233,32 +1233,52 @@ Handle to children.
 @item colormap
 An N-by-3 matrix containing the color map for the current axes.
 @item paperorientation
 @item createfcn
 @item currentaxes 
 Handle to graphics object of current axes.
 @item currentcharacter
 @item currentobject
-@item current_point
+@item currentpoint
+Holds the coordinates of the point over which the mouse pointer was when
+the mouse button was pressed. If a mouse callback function is defined,
+@code{"currentpoint"} holds the coordinates of the point over which the
+mouse pointer is when the function gets called.
 @item deletefcn
 @item dockcontrols 
 --- Values: "on," "off"
 @item doublebuffer 
 --- Values: "on," "off"
 @item filename
 @item handlevisibility 
 --- Values: "on," "off"
 @item hittest
 @item integerhandle
 @item interruptible 
 --- Values: "on," "off"
 @item inverthardcopy
 @item keypressfcn
+see @code{"keypressfcn"}
 @item keyreleasefcn
+With @code{"keypressfcn"}, The keyboard callback functions. These
+callback functions get called when a key is pressed/released
+respectively. The functions are called with two input arguments. The
+first argument holds the handle of the calling figure. The second
+argument holds the event structure which has the following members:
+@table @code
+@item Character
+The ASCII value of the key
+@item Key
+lowercase value of the key
+@item Modifier
+A cell array containing strings representing the modifiers pressed with
+the key. Possible values are @code{"shift"}, @code{"alt"}, and
+@code{"control"}.
+@end table
 @item menubar
 @item mincolormap
 @item name
 @item nextplot
 May be one of
 @table @code
 @item "new"
 @item "add"
@@ -1289,18 +1309,25 @@ Indicates the orientation for printing. 
 @item tag
 @item toolbar
 @item type
 @item units
 @item userdata
 @item visible
 Either @code{"on"} or @code{"off"} to toggle display of the figure.
 @item windowbuttondownfcn
+See @code{"windowbuttonupfcn"}
 @item windowbuttonmotionfcn
+See @code{"windowbuttonupfcn"}
 @item windowbuttonupfcn
+With @code{"windowbuttondownfcn"} and @code{"windowbuttonmotionfcn"},
+The mouse callback functions. These callback functions get called when
+the mouse button is pressed, dragged, and released respectively. When
+these callback functions are called, the @code{"currentpoint"} property
+holds the current coordinates of the cursor.
 @item windowbuttonwheelfcn
 @item windowstyle
 @item wvisual
 @item wvisualmode
 @item xdisplay
 @item xvisual
 @item xvisualmode
 @end table
@@ -1339,16 +1366,20 @@ Setting this property also forces the co
 property to be set to @code{"manual"}.
 @item climmode
 Either @code{"manual"} or @code{"auto"}.
 @item clipping
 @item color
 @item colororder
 @item createfcn
 @item currentpoint
+Holds the coordinates of the point over which the mouse pointer was when
+the mouse button was pressed. If a mouse callback function is defined,
+@code{"currentpoint"} holds the coordinates of the point over which the
+mouse pointer is when the function gets called.
 @item dataaspectratio
 A two-element vector specifying the relative height and width of the
 data displayed in the axes.  Setting @code{dataaspectratio} to @samp{1,
 2]} causes the length of one unit as displayed on the y-axis to be the
 same as the length of 2 units on the x-axis.  Setting
 @code{dataaspectratio} also forces the @code{dataaspectratiomode}
 property to be set to @code{"manual"}.
 @item dataaspectratiomode 
